cmd/zconfigd/zconfigd.c

```diff
@@ -0,0 +1,95 @@
+#include <unistd.h>
+#include <CoreFoundation/CoreFoundation.h>
+#include <IOKit/IOKitLib.h>
+#include <mach/mach.h>
+
+#define kNetLundmanZfsZvol		"net_lundman_zfs_zvol"
+
+// globals
+static IONotificationPortRef    gNotifyPort;
+static io_iterator_t            gKextLoadedIter;
+
+void SignalHandler(int sigraised)
+{
+	printf("\nInterrupted\n");
+
+	// Clean up here
+	IONotificationPortDestroy(gNotifyPort);
+
+	if (gKextLoadedIter)
+	{  
+		IOObjectRelease(gKextLoadedIter);
+		gKextLoadedIter = 0;
+	}
+
+	// exit(0) should not be called from a signal handler.  Use _exit(0) instead
+	//
+	_exit(0);
```

Comment:
Calling CFRunLoopStop() from a signal handler seems to work fine in InvariantDisks:
https://github.com/cbreak-black/InvariantDisks/blob/master/InvariantDisks/IDDispatchUtils.cpp#L17
https://github.com/cbreak-black/InvariantDisks/blob/master/InvariantDisks/IDCLI.cpp#L56
https://github.com/cbreak-black/InvariantDisks/blob/master/InvariantDisks/IDCLI.cpp#L102

But I use a GCD Dispatch Source to handle signals, so I have a lot more freedom which functions to cast (they don't have to be async signal safe). It might not work as well in a normal signal handler.